lib/python/qmk/submodules.py

```diff
@@ -0,0 +1,72 @@
+"""Functions for working with QMK's submodules.
+"""
+
+import subprocess
+
+
+def status():
+    """Returns a dictionary of submodules.
+
+    Each entry is a dict of the form:
+
+        {
+            'name': 'submodule_name',
+            'status': None/False/True,
+            'githash': '<sha-1 hash for the submodule>
+        }
+
+    status is None when the submodule doesn't exist, False when it's out of date, and True when it's current
+    """
+    submodules = {}
+    git_cmd = subprocess.run(['git', 'submodule', 'status'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=30, universal_newlines=True)
+
+    import os
+    for line in git_cmd.stdout.split('\n'):
+        if not line:
+            continue
+
+        status = line[0]
+        githash = line[1:41]
+        submodule = line[42:].split()[0]
```

Comment:
I'm not sure I like these magic numbers... and what if the hash changes in the future...
How about splitting at whitespaces, something like this:
`githash, submodule = line[1:].split()[0:2]`

Reply:
Yeah, I wasn't fond of that either. It's not straightforward to parse because the first character of the line is sometimes a space and sometimes not, and I got stuck on that. Looking again with fresh eyes it does make more sense exclude that first character when splitting.

I also made it more obvious what the if block below that is for.